website/docs/components/table/partials/code/how-to-use.md

```diff
@@ -545,6 +545,153 @@ In this case the table layout is still set to `auto` (default). If instead you w
+    @onPageSizeChange={{this.demoOnPageSizeChange}}
+    @ariaLabel="Pagination for multi-select table"
+  />
+</div>
+```
+
+Depending on the expected behavior, you will need to implement the consumer-side logic that handles the persistence (or not) using the `@onSelectionChange` callback function. For the example above, something like this:
+
+```javascript
+@action
+demoOnSelectionChangeWithPagination({ selectableRowsStates }) {
+  // we loop over all the displayed table rows (a subset of the dataset)
+  selectableRowsStates.forEach((row) => {
+    // we find the record in the dataset corresponding to the current row
+    const recordToUpdate = this.demoSourceData.find(
+      (modelRow) => modelRow.id === row.selectionKey
+    );
+    if (recordToUpdate) {
+      // we update the record `isSelected` state based on the row (checkbox) state
+      recordToUpdate.isSelected = row.isSelected;
+    }
+  });
+}
+
+```
+
+For details about the arguments provided to the `@onSelectionChange` callback function refer to the [Component API](#component-api) section.
+
+
+#### Usability and accessibility considerations
+
+Since the "selected" state of a row is communicated visually via the checkbox selection and for screen-reader users via the `aria-label` applied to the checkbox, there are some important considerations to keep in mind when implementing a multi-select table.
+
+If the selection status of the rows is persisted even when a row is not displayed in the UI, you need to be mindful of what the expectations of the user may be: how do we make them aware that the action that they are going to perform may involve table rows that were previously selected by the user but that are not displayed in that moment?
+
+Even more complex is the case of the "Select all" checkbox. While it's pretty straightforward the expected behaviour when all the table rows are displayed on the table (precisely, "select all the rows in the table"), it may not be obvious what is the expected behaviour when the table rows are paginated of filtered.
+
+If as a user I am looking at a subset of the whole possible rows, and I click "select all", what should I expect to happen? That all (and only) the displayed rows are selected –and this is what happens in the example above– or that all the data/model rows are selected, even if not displayed at the moment? In the first case, the "select all" state changes whenever the rows are changes (eg. the user chooses a different pagination number), and this could be confusing. In the second case, it may not be obvious to the user that _all_ the data records have been selected, and it may unintentionally perform an action on them under the assumption that they are performing it only on the displayed table rows.
+
+Whatever functionality you decide to implement, be mindful of all these possible subtleties and complexities.
+
+For more detailed guidance around these usability and accessibility concerns, refer to **[TODO!!!]**.
```

Comment:
@hashibot-hds what should I do here? not sure what I should refer to (we don't have specific documentation about this). Should I say "contact the HDS team" for the time being, until we have documented guidance that I can link to?

Reply:
Maybe this is where something is included like:

> At a bare minimum we recommend clearly communicating to the user if they have selected rows outside of their current view and how many out of the total data set are selected. We're working to document these scenarios as they arise, in the meantime [contact the HDS team](/about/support) for assistance.

Reply:
I think it makes sense to ask them to contact us for guidance. Once we have more detailed guidance, we will just update this to include it.

(I will edit this and push up.)